spec/models/user_spec.rb

```diff
@@ -117,6 +133,39 @@
         expect(User.not_in_experience).to contain_exactly(user_not_in_experience)
       end
     end
+
+    describe '#by_roles_in' do
+      let!(:user1) { create(:user, roles: %i[admin super_admin]) }
+      let!(:user2) { create(:user, roles: [:super_admin]) }
+
+      subject { User.by_roles_in(roles_names).to_a }
```

Comment:
the subject should be only the build of the object being tested, not the result of the exercised method, this should be  in the test itself